### ISSUE TITLE
DataDictionary.Empty is not protected by potential bugs.

### ISSUE THREAD

**Issue by frabe1579** [FoundatioFx/Foundatio#280]
I've found a potential source of bugs:
the class `Foundatio.Utility.DataDictionary` exposes a static readonly field `Empty` to represent empty dictionaries, but it's instance is not readonly, so values can be added to `Empty` causing potential bugs.
The current declaration is: 
```
public static readonly DataDictionary Empty = new();
```
A new safe declaration shoud be
```
public static DataDictionary Empty => new();
```
This at least returns always empty dictionaries, but creates a new instance at every call.
Another approach should be to re-implement DataDictionary with explicit implementation of IDictionary and IReadOnlyDictionary, or something else.

**Comment by niemyjski** [comment]
Yes, we should work on making empty be read-only. @ejsmith thoughts on above, kind of would be nice to make it truly readonly instead of returning a new instance. But would take a little bit of work.

**Comment by frabe1579** [comment]
Maybe a `IReadOnlyDictionary<,>`, with appropriate extension methods, should be used when exposed for example on `IQueueEntry<>`, but I don't know if it's used in other code.

**Comment by niemyjski** [comment]
Thanks for the detailed write-up. This is a real concern, but the fix is more involved than it might appear.

DataDictionary extends Dictionary<string, object?>, which is mutable by design. We can't simply return a singleton Empty instance and protect it — any caller can mutate it. The safe approaches are:

1. Return a new instance per call (=> new()) — avoids the shared-state bug but allocates on every access, which defeats the point of a static Empty.
2. Introduce a frozen/read-only wrapper — we'd need to subclass DataDictionary, override every mutating method (Add, Remove, this[key] = ..., Clear) to throw InvalidOperationException, and return that as the singleton. This is the pattern .NET itself uses for Array.Empty<T>() and ReadOnlyCollection<T>, but it's non-trivial to get right for a dictionary.
3. Change the exposed type to IReadOnlyDictionary<string, object?> — the cleanest API boundary, but it's a breaking change on anything that currently accepts or assigns a DataDictionary directly (e.g., IQueueEntry<T>.Properties).

For now the safest non-breaking fix is option 2 — a private FrozenDataDictionary subclass that throws on mutation. We'd be open to a PR along those lines if you're interested in tackling it. Option 3 (the IReadOnlyDictionary surface change) is worth a separate discussion as it touches public API contracts.
